engine/src/cmd/collide_map.cpp

```diff
@@ -161,8 +160,8 @@ void CollideArray::flatten() {
             collideUpdate(*tmp, index);
         }
 
-        std::list<CollidableBackref>::iterator listend = toflattenhints[i].end();
-        for (std::list<CollidableBackref>::iterator j = toflattenhints[i].begin();
+        auto listend = toflattenhints[i].end();
+        for (auto j = toflattenhints[i].begin();
                 j != listend;
```

Comment:
we should just test directly against `.end()` instead of creating a variable here.

Reply:
NOTE: the `end` could potentially change which is why you're always supposed to test directly against `end()` instead of using a var

Reply:
@BenjamenMeyer you're probably right. I would like to point out that the `.end()` pre-assignment was there before; I just changed the variable type declarations to `auto`, based on CLion's suggestion to use `auto` for all iterator variables. Based on @royfalk 's comment further down, it sounds like we still need to discuss whether or not we want to do that as a general practice. Maybe I'll revert the changes to these lines for now.

Reply:
yeah,  I realize it was there before; this would be one improvement for the code changing that up to actually be correct. Use `auto` instead of specifying the full iterator type is also useful here (and one of few things I like to use `auto` for).

Reply:
OK, I'll change this to use `.end()` directly. I'll also leave it as `auto` for now. Any objections?

Reply:
I think this is all that this PR is waiting on for approval

Reply:
Fixed.